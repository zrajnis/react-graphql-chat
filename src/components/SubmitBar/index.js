import React from 'react'

import Button from 'components/Button'
import Input from 'components/Input'
import ErrorMsg from 'components/ErrorMsg'

const SubmitBar = ({ inputVal, label, legend, handleChange, handleSubmit, errorMsg, buttonText }) => {
  return (
    <form onSubmit={handleSubmit}>
      <fieldset>
        <legend> {legend} </legend>
        <Input id='textField' label={label} type='text' value={inputVal}
          onChange={handleChange} />
        <Button id='sendBtn' type='submit'> {buttonText} </Button>
        {errorMsg && <ErrorMsg > {errorMsg} </ErrorMsg>}
      </fieldset>
    </form>
  )
}

export default SubmitBar
