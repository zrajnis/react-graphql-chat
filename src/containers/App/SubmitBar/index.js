import React from 'react'

import Button from 'components/Button'
import Input from 'components/Input'

const SubmitBar = () => {
  return (
    <div>
      <Input id="textField" label="Insert a text message" type="text" />
      <Button id="sendBtn"> Send </Button>
    </div>
  )
}

export default SubmitBar
