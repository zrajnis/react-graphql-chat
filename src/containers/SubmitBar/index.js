import React from 'react'

import Button from 'components/Button'
import Input from 'components/Input'

const SubmitBar = ({ inputVal, handleChange, handleSubmit }) => {
  return (
    <div>
      <form onSubmit={handleSubmit}>
        <Input id="textField" label="Insert a text message" type="text" value={inputVal} onChange={handleChange} />
        <Button id="sendBtn" type="submit"> Send </Button>
      </form>
    </div>
  )
}

export default SubmitBar
