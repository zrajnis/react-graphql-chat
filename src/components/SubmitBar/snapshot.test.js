import React from 'react'
import renderer from 'react-test-renderer'
import SubmitBar from 'components/SubmitBar'

describe('Submit bar component snapshot', () => {
  const mockFn = jest.fn()

  it('renders the submit bar', () => {
    const tree = renderer.create(
      <SubmitBar
        buttonText='Submit'
        error
        errorMsg='test errorMsg'
        handleChange={mockFn}
        handleSubmit={mockFn}
        inputVal='testVal'
        label='test label'
        legend='test legend'
      />
    ).toJSON()

    expect(tree).toMatchSnapshot()
  })
})
