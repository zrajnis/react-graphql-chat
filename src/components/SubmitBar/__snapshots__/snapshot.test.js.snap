// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Submit bar component snapshot renders the submit bar 1`] = `
<form
  onSubmit={undefined}
>
  <fieldset>
    <legend>
       
       
    </legend>
    <div
      className="input-container--316X5DI-"
    >
      <label
        htmlFor="textField"
      >
         
        test label
         
      </label>
      <input
        className="input--3L2xwewm"
        id="textField"
        onChange={undefined}
        type="text"
        value="testVal"
      />
    </div>
    <button
      className="button--3j7htfnZ"
      id="sendBtn"
      type="submit"
    >
       
      Submit
       
    </button>
    <div
      className="error-msg--Gu7cE9b_"
    >
       
       
      test errorMsg
       
       
    </div>
  </fieldset>
</form>
`;
